libs/wire-api/src/Wire/API/User.hs

```diff
@@ -872,35 +866,29 @@ modelChangePassword = Doc.defineModel "ChangePassword" $ do
   Doc.property "new_password" Doc.string' $
     Doc.description "New password (6 - 1024 characters)"
 
-instance ToJSON PasswordChange where
-  toJSON (PasswordChange old new) =
-    A.object
-      [ "old_password" A..= old,
-        "new_password" A..= new
-      ]
-
-instance FromJSON PasswordChange where
-  parseJSON = A.withObject "PasswordChange" $ \o ->
-    PasswordChange
-      <$> o A..:? "old_password"
-      <*> o A..: "new_password"
+instance ToSchema PasswordChange where
+  schema =
+    object "PasswordChange" $
+      PasswordChange
+        <$> cpOldPassword .= maybe_ (optField "old_password" schema)
+        <*> cpNewPassword .= field "new_password" schema
 
 newtype LocaleUpdate = LocaleUpdate {luLocale :: Locale}
   deriving stock (Eq, Show, Generic)
   deriving newtype (Arbitrary)
+  deriving (ToJSON, FromJSON, S.ToSchema) via (Schema LocaleUpdate)
 
 modelChangeLocale :: Doc.Model
 modelChangeLocale = Doc.defineModel "ChangeLocale" $ do
   Doc.description "Data to change a locale."
   Doc.property "locale" Doc.string' $
     Doc.description "Locale to be set"
 
-instance ToJSON LocaleUpdate where
-  toJSON l = A.object ["locale" A..= luLocale l]
-
-instance FromJSON LocaleUpdate where
-  parseJSON = A.withObject "locale-update" $ \o ->
-    LocaleUpdate <$> o A..: "locale"
+instance ToSchema LocaleUpdate where
+  schema =
+    object "locale-update" $
```

Comment:
Better to use camel case object names, for consistency in the swagger documentation.